src/components/customCursor/customCursor.tsx

```diff
@@ -0,0 +1,64 @@
+'use client';
+import { useEffect } from 'react';
+import gsap from 'gsap';
+
+const CustomCursor = () => {
+  useEffect(() => {
+    // CursorFollow logic
+    const moveCursor = (e: MouseEvent) => {
+      gsap.to('.cursorFollower', {
+        x: e.clientX - 6,
+        y: e.clientY - 6,
+        duration: 1.5,
+        ease: 'power4.out',
+      });
+      gsap.to('.cursor', {
+        x: e.clientX - 4,
+        y: e.clientY - 4,
+        duration: 0.5,
+        ease: 'power4.out',
+      });
+    };
+
+    document.addEventListener('mousemove', moveCursor);
+
+    // Hover logic
+    const mouseEnter = () => {
+      gsap.set('.cursor', { display: 'none' });
+      gsap.to('.cursorFollower', {
+        scale: 2,
+      });
+    };
+
+    const mouseLeave = () => {
+      gsap.set('.cursor', { display: 'block' });
+      gsap.to('.cursorFollower', {
+        scale: 1,
+      });
+    };
+
+    document.querySelectorAll('.hovering').forEach((el) => {
+      el.addEventListener('mouseenter', mouseEnter);
+      el.addEventListener('mouseleave', mouseLeave);
+    });
```

Comment:
event listeners should be removed before the next rerender, in the cleanup function

Reply:
done